r.in.gdal/r.external: add basic support for GDALs GDT_Int8

ninsbl opened the pull request:
This PR adds basic support for importing GDT_Int8 datatype via GDAL.

I am no C-programmer, so I would be happy if a knowledgeable person could look into this. I am especially unsure about the _#ifndef_ part (note that _14_ is the constant representing GDT_Int8).

Addresses #4230

rouault commented:
> #if GDAL_VERSION_NUM >= 3000000

Better use
``
#if GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(3,7,0)
``

rouault commented on lib/raster/get_row.c:
```diff
@@ -368,6 +372,9 @@ static void gdal_values_int(int fd, const unsigned char *data,
             c[i] = *(GByte *)d;
             break;
+        case GDT_Int8:
+            c[i] = *(GInt16 *)d;
```
I'm not familiar at all with that code, but I believe this should be:
```suggestion
            c[i] = *(signed char *)d;
```

ninsbl replied:
Included in bc9f6a89890e882a19d8a0e35fb92346d8183c78

nilason replied:
Why not `c[i] = *(GInt8 *)d;` to use the GDAL API and for consistency in the file?

rouault replied:
yes, although I'm not  big fan of our custom type . Using C99's int8_t would probably be better

nilason replied:
Indeed, standard fixed sized integers for fixed sized data types are certainly better.

ninsbl commented:
Thanks @neteler and @rouault ! Your feedback is very much appreciated. I tried to implement yor suggestions in the latest commits...

nilason commented:
Stefan, you should also add `#include <stdint.h>` (which is where the C99 fixed width integer types are defined, [see more](https://en.cppreference.com/w/c/types/integer)) to `get_row.c`.

ninsbl commented:
> Stefan, you should also add `#include <stdint.h>` (which is where the C99 fixed width integer types are defined, [see more](https://en.cppreference.com/w/c/types/integer)) to `get_row.c`.

Thanks for the hint. Added in [1565b1d](https://github.com/OSGeo/grass/pull/4256/commits/1565b1de59640450ca4177adaac2f15eff550fe1)

I also added a test for this case.

ninsbl commented:
Thanks for your approval @nilason . I had to add a switch to skip the test for systems with GDAL < 3.7 (like Ubuntu 22.04, where tests failed). That commit dismissed your review, it would be great if you could re-approve...

nilason commented:
Should this be backported?

nilason commented:
Backported to 8.4 with c4b2304.
